### PR TITLE
zlib: Add support for info: true option

### PR DESCRIPTION
**Pull request by npaun**
This PR implements the `{info: true}` option for zlib. It's not documented for brotli, but the unit tests expect it there too. We return an engine, and the buffer. 

~However, the engine we're returning isn't really legit - we just create it on the fly to return it. Should we fall back to the streaming path when `info: true`, for more fidelity with NodeJS or is this close enough?~ Now going whole-hog and falling back to the streaming implementation when `info: true`

### CODE REVIEW

**Review by anonrig — CHANGES_REQUESTED** [review]
We're diverging from node.js zlib implementation and it is producing different states for `info: true`.

For example, on Node.js

```js
> require('zlib').brotliCompressSync('aaa', { info: true })
{
  buffer: <Buffer 0b 01 80 61 61 61 03>,
  engine: BrotliCompress {
    _writeState: Uint32Array(2) [ 16377, 0 ],
```

but calling initializing BrotliCompress directly produces different results on `writeState`

```js
> require('zlib').BrotliCompress({ info: true })
<ref *1> BrotliCompress {
  _writeState: Uint32Array(2) [ 0, 0 ],
```

File: src/node/internal/zlib.d.ts
Comment (anonrig) [inline src/node/internal/zlib.d.ts:252]: Enum values are not 1 to 1 representably Node.js. We should avoid using them. Can we convert this to an object and move this to the file where it is used?

File: src/node/internal/zlib.d.ts
Comment (anonrig) [inline src/node/internal/zlib.d.ts:5]: The next line cover this already, why did we need this line?

File: src/node/internal/internal_zlib.ts
Comment (anonrig) [inline src/node/internal/internal_zlib.ts:260]: Nit: Can we change the variables names? wdyt?

```suggestion
  zlibImpl(CONST_GZIP, data, options, callback);
```

File: src/node/internal/internal_zlib.ts
Comment (anonrig) [inline src/node/internal/internal_zlib.ts:174]: Technically this is possible, but it will only be possible if we have a bug on our implementation right? In order to find the bugs easier, what about using `.cause`?

```suggestion
    const unreachable = new Error('Unreachable')
    unreachable.cause = err;
    throw unreachable;
```

File: src/node/internal/zlib.d.ts
Comment (npaun) [reply src/node/internal/zlib.d.ts:5]: Must've been a copy paste accident; nuked

File: src/node/internal/zlib.d.ts
Comment (npaun) [reply src/node/internal/zlib.d.ts]: I moved this enum to internal_zlib.ts, but I was not sure how to replace it with an object, because `keyof typeof CLASS_BY_MODE` is just `number`.

**Review by anonrig — APPROVED** [review]
Let's wait for @jasnell to review, before landing.

**Review by jasnell — APPROVED** [review]

**Pull request merged**
